spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -1115,6 +1119,7 @@ object CometSparkSessionExtensions extends Logging {
         BinaryType | StringType | _: DecimalType | DateType | TimestampType =>
       true
     case t: DataType if t.typeName == "timestamp_ntz" => true
+    case s: StructType => isSchemaSupported(s)
```

Comment:
Is this change needed? In general, structs are not supported (yet) and since this method is not operator specific, we probably shouldn't have this here.

Reply:
See https://github.com/apache/datafusion-comet/pull/731#discussion_r1698972618, it's still needed by shouldApplyRowToColumnar

Reply:
Hmm. I'm concerned that someone might use this method and get unexpected behavior. A comment to explain why this is here would be justified, I think?

Reply:
Yeah it probably makes sense just to move it into CometRowToColumnarExec